From e16e14d6da205fe1fcd3b20f8ed0ec5aaca6456b Mon Sep 17 00:00:00 2001
From: Saurabh Sengar <saurabh.singh@xilinx.com>
Date: Fri, 2 Mar 2018 07:11:43 +0530
Subject: [PATCH 0312/1851] drm: xlnx: mixer: Correcting loop termination
 condition

commit 4621d085b4b6778689be5f12665528ed1123a6ff from
https://github.com/Xilinx/linux-xlnx.git

The layer id starts from 0 and max id should be one less then total count.
The loop was iterating till mixer->layer_cnt, so correcting it to one less.

Signed-off-by: Saurabh Sengar <saurabh.singh@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_mixer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_mixer.c b/drivers/gpu/drm/xlnx/xlnx_mixer.c
index e4a05c5a14a5..f0a3d540663e 100644
--- a/drivers/gpu/drm/xlnx/xlnx_mixer.c
+++ b/drivers/gpu/drm/xlnx/xlnx_mixer.c
@@ -2500,7 +2500,7 @@ static void xlnx_mix_init(struct xlnx_mix_hw *mixer)
 	/* default to blue */
 	xlnx_mix_set_bkg_col(mixer, rgb_bg_clr);
 
-	for (i = 0; i <= mixer->layer_cnt; i++) {
+	for (i = 0; i < mixer->layer_cnt; i++) {
 		layer_id = mixer->layer_data[i].id;
 		layer_data = &mixer->layer_data[i];
 		if (layer_id == XVMIX_LAYER_MASTER)
-- 
2.31.1

